cmd/admin-service-stop.go

```diff
@@ -0,0 +1,98 @@
+/*
+ * Minio Client (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"net/url"
+
+	"github.com/minio/cli"
+	"github.com/minio/minio/pkg/madmin"
+	"github.com/minio/minio/pkg/probe"
+)
+
+var (
+	adminServiceStopFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all",
+			Usage: "Control all nodes in the cluster",
+		},
+	}
+)
+
+var adminServiceStopCmd = cli.Command{
+	Name:   "stop",
+	Usage:  "Stop a minio server",
+	Action: mainAdminServiceStop,
+	Flags:  append(adminServiceStopFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+   mc admin service {{.Name}} - {{.Usage}}
+
+USAGE:
+   mc admin service {{.Name}} ALIAS
+
+FLAGS:
+  {{range .Flags}}{{.}}
+  {{end}}
+EXAMPLES:
+`,
+}
+
+// checkAdminServiceStopSyntax - validate all the passed arguments
+func checkAdminServiceStopSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 2 {
+		cli.ShowCommandHelpAndExit(ctx, "stop", 1) // last argument is exit code
+	}
+}
+
+func mainAdminServiceStop(ctx *cli.Context) error {
+
+	setGlobalsFromContext(ctx)
+	checkAdminServiceStopSyntax(ctx)
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Fetch the server's address stored in config
+	_, _, hostCfg, perr := expandAlias(aliasedURL)
+	if perr != nil {
+		return perr.ToGoError()
+	}
+
+	// Check if alias exists
+	if hostCfg == nil {
+		fatalIf(errInvalidAliasedURL(aliasedURL).Trace(aliasedURL), "The specified alias is not found.")
+	}
+
+	// Parse the server address
+	url, err := url.Parse(hostCfg.URL)
+	if err != nil {
+		return err
+	}
+
+	// Create a new Minio Admin Client
+	client, err := madmin.New(url.Host, hostCfg.AccessKey, hostCfg.SecretKey, url.Scheme == "https")
+	if err != nil {
+		return err
+	}
+
+	// Stop the specified Minio server
+	err = client.ServiceStop()
```

Comment:
Because probe.NewError is used as 'err' and golang err is used as 'e' this is to differentiate we have consistently followed everywhere.